lib/components/ImageButton.tsx

```diff
@@ -0,0 +1,226 @@
+  /** Parameter `icon` of component `DmIcon`. */
+  dmIcon: string | null;
+  /** Parameter `icon_state` of component `DmIcon`. */
+  dmIconState: string | null;
+  /** 
+   * Changes the layout of the button, making it fill the entire horizontally available space.
+   * Allows the use of `title` 
+   */
+  fluid: boolean;
+  /** Parameter responsible for the size of the image, component and standard "stubs". */
+  imageSize: number;
+  /** Prop `src` of Image component. Example: `imageSrc={resolveAsset(thing.image}` */
+  imageSrc: string;
+  /** Called when button is clicked with LMB. */
+  onClick: (e: any) => void;
+  /** Called when button is clicked with RMB. */
+  onRightClick: (e: any) => void;
+  /** Makes button selected and green if true. */
+  selected: BooleanLike;
+  /** Requires `fluid` for work. Bold text with divider betwen content. */
+  title: string;
+  /** A fancy, boxy tooltip, which appears when hovering over the button */
+  tooltip: ReactNode;
+  /** Position of the tooltip. See [`Popper`](#Popper) for valid options. */
+  tooltipPosition: Placement;
+}> &
+  BoxProps;
+
+export function ImageButton(props: Props) {
+  const {
+    asset,
+    base64,
+    buttons,
+    buttonsAlt,
+    children,
+    className,
+    color,
+    disabled,
+    dmFallback,
+    dmIcon,
+    dmIconState,
+    fluid,
+    imageSize = 64,
+    imageSrc,
+    onClick,
+    onRightClick,
+    selected,
+    title,
+    tooltip,
+    tooltipPosition,
+    ...rest
+  } = props;
+
+  function getFallback(iconName: string, iconSpin: boolean) {
+    return (
+      <Stack height={`${imageSize}px`} width={`${imageSize}px`}>
+        <Stack.Item grow textAlign="center" align="center">
+          <Icon
+            spin={iconSpin}
+            name={iconName}
+            color="gray"
+            style={{ fontSize: `calc(${imageSize}px * 0.75)` }}
+          />
+        </Stack.Item>
+      </Stack>
+    );
+  }
+
+  let buttonContent = (
+    <div
+      className={classes([
+        styles.container,
+        buttons && styles.hasButtons,
+        !onClick && !onRightClick && styles.noAction,
+        selected && styles.selected,
+        disabled && styles.disabled,
+        color && typeof color === 'string'
+          ? styles['color__' + color]
+          : styles['color__default'],
+      ])}
+      tabIndex={!disabled ? 0 : undefined}
+      onClick={(event) => {
+        if (!disabled && onClick) {
+          onClick(event);
+        }
+      }}
+      onContextMenu={(event) => {
+        event.preventDefault();
+        if (!disabled && onRightClick) {
+          onRightClick(event);
+        }
+      }}
+      style={{ width: !fluid ? `calc(${imageSize}px + 0.5em + 2px)` : 'auto' }}
+    >
+      <div className={classes([styles.image])}>
+        {base64 || asset || imageSrc ? (
+          <Image
+            className={classes(!base64 && !imageSrc && asset || [])}
+            src={base64 ? `data:image/jpeg;base64,${base64}` : imageSrc}
```

Comment:
jpeg? pretty sure `data:image/png;base64,` is used everywhere

Reply:
Nah, everywhere is jpeg, there is no differense, transparency allowed anyway

Reply:
> everywhere is jpeg

this is just patently untrue

> transparency allowed anyway

i don't believe we should be serving wrong MIME types intentionally

Reply:
> > everywhere is jpeg
> 
> this is just patently untrue
> 
> > transparency allowed anyway
> 
> i don't believe we should be serving wrong MIME types intentionally

![image](https://github.com/user-attachments/assets/cc8f9476-b830-49f1-b47e-f7e80af01d99)

Reply:
yes, but `tgui-core` is not just used by tgstation
![image](https://github.com/user-attachments/assets/63e37670-6190-48aa-9049-95cfd76c71b7)